src/js/port_handler.js

```diff
@@ -200,13 +200,22 @@ PortHandler.update_port_select = function (ports) {
 
     for (let i = 0; i < ports.length; i++) {
         let portText;
+        let opt;
+        let str;
         if (ports[i].displayName) {
             portText = (`${ports[i].path} - ${ports[i].displayName}`);
+            str = `${ports[i].displayName}`;
+            if (str.includes('STM32')) {
+                opt = `<option selected value='${ports[i].path}'>`;
+            } else {
```

Comment:
I think the selection of port must be done outside the generation. In this way we can decide if they are several STM32, apply a more complex method, let the generation cleaner, only call it if necessary, etc. 

We usually we use jquery with it, it's cleaner, with a simple: `this.portPickerElement.val(value_to_select).change()` will be enough in both cases.

This is being done at some places of code, for example this is called just after the method you have changed, so better add your logic this:

https://github.com/betaflight/betaflight-configurator/blob/e8b097d6c59a56ffaed6bb7f19490a8bdb85bc92/src/js/port_handler.js#L122-L126

Other place:

https://github.com/betaflight/betaflight-configurator/blob/e8b097d6c59a56ffaed6bb7f19490a8bdb85bc92/src/js/port_handler.js#L92

Other place:

https://github.com/betaflight/betaflight-configurator/blob/e8b097d6c59a56ffaed6bb7f19490a8bdb85bc92/src/js/port_handler.js#L178

You can add it in the if of the first sample, or make a more complex method called from all of this places.

Reply:
Thanks. Working on it!

Reply:
Line 122 in port_handler.js is already part of finding a new port (after initialization is done or perhaps not yet complete).
Refactored it into a function. I think it's preferred to use last_user_port.